File: apple/Sources/ReactNativeWatchOS/Modifiers/FontWeight.swift
import SwiftUI

/// SwiftUI `.fontWeight(_:)`. Sets the stroke weight of text within the
/// view. An unrecognized (or absent) weight leaves the view unchanged.
enum RNWFontWeightModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("fontWeight") { view, params, _ in
            guard let weight = RNWFontParsers.weight(params.string("weight")) else {
                return view
            }
            return AnyView(view.fontWeight(weight))
        }
    }
}

File: src/modifiers/fontWeight.ts
import { createModifier } from './createModifier';
import type { FontWeight } from './font';

export interface FontWeightParams {
  weight?: FontWeight;
}

/// SwiftUI `.fontWeight(_:)`. Sets the stroke weight of text within the
/// view. Pass a weight string directly or an object.
export function fontWeight(params?: FontWeightParams | FontWeight) {
  if (typeof params === 'string') {
    return createModifier('fontWeight', { weight: params });
  }
  return createModifier('fontWeight', params);
}
